tutorials/analyze-cryptocurrency-data.md

```diff
@@ -0,0 +1,624 @@
+   time            TIMESTAMP WITH TIME ZONE NOT NULL,
+   opening_price   DOUBLE PRECISION,
+   highest_price   DOUBLE PRECISION,
+   lowest_price    DOUBLE PRECISION,
+   closing_price   DOUBLE PRECISION,
+   volume_eth      DOUBLE PRECISION,
+   volume_currency DOUBLE PRECISION,
+   currency_code   VARCHAR (10)
+);
+
+--Timescale specific statements to create hypertables for better performance
+SELECT create_hypertable('btc_prices', 'time');
+SELECT create_hypertable('eth_prices', 'time');
+SELECT create_hypertable('crypto_prices', 'time');
+```
+
+Note that we include 3 `create_hypertable` statements which are special TimescaleDB 
+statements. A hypertable is an abstraction of a single continuous table across 
+time intervals, such that one can query it via vanilla SQL. For more on hypertables, 
+see the [Timescale docs][hypertable-docs] and this [blog post][hypertable-blog].
+
+### Step 2: Create a dataset to analyze
+
+Now that we’ve defined the data we want, it’s time to construct a dataset containing that 
+data. To do this, we’ll write a small Python script for extracting data from [CryptoCompare][cryptocompare] 
+into 4 CSV files (`coin_names.csv`, `crypto_prices.csv`, `btc_prices.csv`, and `eth_prices.csv`).
+
+In order to get data from CryptoCompare, you’ll need to [obtain an API key][cryptocompare-apikey]. 
+For this analysis, the free key should be plenty.
+
+The script consists of five parts:
+
+- Importing the necessary Python libraries in order to complete the data extraction
+- Populate the `currency_info` table with a list of coin names
+- Get the historical Bitcoin (BTC) prices in 4198 other cryptocurrencies and populate the `crypto_prices` table
+- Get historical Bitcoin prices in different fiat currencies to populate `btc_prices`
+- Get historical Ethereum prices in different fiat currencies to populate `eth_prices`
+
+Here's the full Python script, which you can also [:DOWNLOAD_LINK: download][dataset-creation]:
+
+```python
+#####################################################################
+#1. Import library and setup API key
+#####################################################################
+import requests
+import json
+import csv
+from datetime import datetime
+
+apikey = 'YOUR_CRYPTO_COMPARE_API_KEY'
+#attach to end of URLstring
+url_api_part = '&api_key=' + apikey
+
+#####################################################################
+#2. Populate list of all coin names
+#####################################################################
+#URL to get a list of coins from cryptocompare API
+URLcoinslist = 'https://min-api.cryptocompare.com/data/all/coinlist'
+
+#Get list of cryptos with their symbols
+res1 = requests.get(URLcoinslist)
+res1_json = res1.json()
+data1 = res1_json['Data']
+symbol_array = []
+cryptoDict = dict(data1)
+
+#write to CSV
+with open('coin_names.csv', mode = 'w') as test_file:
+   test_file_writer = csv.writer(test_file, delimiter = ',', quotechar = '"', quoting=csv.QUOTE_MINIMAL)
```

Comment:
This is slightly long, requiring scrolling in the code block.  Consider formatting to reduce length

<img width="925" alt="Screen Shot 2020-03-02 at 5 26 06 PM" src="https://user-images.githubusercontent.com/12288273/75723827-0fa73200-5cab-11ea-9033-3fc517bee1a6.png">